packages/react/src/template.tsx

```diff
@@ -176,35 +177,30 @@ export function BaseTemplate({
             key: parentNode.key,
             component: parentNode.component,
             propKey,
-            storeKey: 'ref' in parentNode ? extractRefKey(parentNode.ref) : storeKey,
+            path,
           },
-          dataState: stateNode,
+          dataState: propValue,
           payload,
         })
       }
     }
-    if (Array.isArray(stateNode)) {
-      return stateNode.map((item, index) =>
-        renderProp(propKey, item, parentNode, `${path}.props[${propKey}][${index}]`)
-      )
+    if (Array.isArray(propValue)) {
+      return propValue.map((item) => renderProp(propKey, item, parentNode, path))
     }
-    if (isCompositeDataState(stateNode)) {
-      return render(stateNode, `${path}.props[${propKey}]`)
+    if (isCompositeDataState(propValue)) {
+      return render(propValue, path)
     }
-    if (stateNode && typeof stateNode === 'object') {
+    if (propValue && typeof propValue === 'object') {
       return Object.fromEntries(
-        Object.entries(stateNode).map(([key, value]) => {
-          return [
-            key,
-            renderProp(`${propKey}.${key}`, value, parentNode, `${path}.props[${propKey}][${key}]`),
-          ]
+        Object.entries(propValue).map(([key, value]) => {
+          return [key, renderProp(key, value, parentNode, path)]
```

Comment:
Nope! That was for computing those keys. So, if there was a component passed via a prop, an auto-generated key would include "props[key]". We no longer generate keys.

Path, in this context, is just path to the store that data is coming from. By default, its "''", but if we're on the liveMedia, it will be "liveMedia:layers:${hash}" (just an example).

Reply:
Ok but the path to this exact data value is `[...path, 'props', key]`

But maybe this isn't an issue. I'd have to play with things to see if this matters at all